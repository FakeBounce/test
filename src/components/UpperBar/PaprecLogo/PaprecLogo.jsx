import React, { Component } from "react";
import styles from "./PaprecLogo.module.scss";

import Logo from "assets/images/top-right-logo.png";
import { Media } from "reactstrap";
import { translate } from "common/methods/translations";
import { observer } from "mobx-react";
import { observable } from "mobx";

@observer
class PaprecLogo extends Component {
  @observable currentLanguage = localStorage.getItem("language");
  @observable tmp = translate("globals.header.help");

  render() {
    const { prapesHref } = this.props;
    return (
      <div className={styles.ColorImageContainer}>
        <a href={prapesHref} target="_blank">
          <Media className={styles.TopRightLogo} src={Logo} />
        </a>
      </div>
    );
  }
}
export default PaprecLogo;
