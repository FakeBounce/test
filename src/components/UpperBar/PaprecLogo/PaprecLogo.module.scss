.ColorImageContainer {
  order: 6;
  width: 50px;
  background-color: #e3672a;
  align-items: center;
  display: flex;
  justify-content: center;
}