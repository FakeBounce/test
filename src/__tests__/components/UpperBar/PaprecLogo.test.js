import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import PaprecLogo from "components/UpperBar/PaprecLogo/PaprecLogo";

configure({ adapter: new Adapter() });

describe("Testing PaprecLogo component", () => {
  it("renders as expected without param", () => {
    const wrapper = shallow(<PaprecLogo />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders as expected with param", () => {
    const wrapper = shallow(<PaprecLogo prapesHref={"https://google.fr"} />);
    expect(wrapper).toMatchSnapshot();
  });
});
