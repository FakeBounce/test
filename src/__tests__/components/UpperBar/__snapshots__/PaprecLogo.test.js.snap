// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PaprecLogo component renders as expected with param 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaprecLogo
    prapesHref="https://google.fr"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        href="https://google.fr"
        target="_blank"
>
        <Media
                className={undefined}
                src="top-right-logo.png"
        />
</a>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Media
          className={undefined}
          src="top-right-logo.png"
/>,
        "href": "https://google.fr",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": undefined,
          "src": "top-right-logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href="https://google.fr"
          target="_blank"
>
          <Media
                    className={undefined}
                    src="top-right-logo.png"
          />
</a>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Media
            className={undefined}
            src="top-right-logo.png"
/>,
          "href": "https://google.fr",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": undefined,
            "src": "top-right-logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing PaprecLogo component renders as expected without param 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaprecLogo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        href={undefined}
        target="_blank"
>
        <Media
                className={undefined}
                src="top-right-logo.png"
        />
</a>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Media
          className={undefined}
          src="top-right-logo.png"
/>,
        "href": undefined,
        "target": "_blank",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": undefined,
          "src": "top-right-logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href={undefined}
          target="_blank"
>
          <Media
                    className={undefined}
                    src="top-right-logo.png"
          />
</a>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Media
            className={undefined}
            src="top-right-logo.png"
/>,
          "href": undefined,
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": undefined,
            "src": "top-right-logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
